src/R/Editor/Impl/Completion/Providers/FilesCompletionProvider.cs

```diff
@@ -64,33 +69,78 @@ public class FilesCompletionProvider : IRCompletionListProvider {
                     _cachedUserDirectory = userDirectory;
                     directory = Path.Combine(userDirectory, _directory);
                 } else {
-                    directory = Path.Combine(RToolsSettings.Current.WorkingDirectory, _directory);
+                    directory = _directory;
                 }
 
-                if (Directory.Exists(directory)) {
-                    foreach (string dir in Directory.GetDirectories(directory)) {
-                        DirectoryInfo di = new DirectoryInfo(dir);
-                        if (!di.Attributes.HasFlag(FileAttributes.Hidden) && !di.Attributes.HasFlag(FileAttributes.System)) {
-                            string dirName = Path.GetFileName(dir);
-                            completions.Add(new RCompletion(dirName, dirName + "/", string.Empty, folderGlyph));
-                        }
-                    }
+                if (!string.IsNullOrEmpty(directory)) {
+                    IEnumerable<RCompletion> entries = null;
 
-                    foreach (string file in Directory.GetFiles(directory)) {
-                        FileInfo di = new FileInfo(file);
-                        if (!di.Attributes.HasFlag(FileAttributes.Hidden) && !di.Attributes.HasFlag(FileAttributes.System)) {
-                            ImageSource fileGlyph = _imagesProvider?.GetFileIcon(file);
-                            string fileName = Path.GetFileName(file);
-                            completions.Add(new RCompletion(fileName, fileName, string.Empty, fileGlyph));
-                        }
+                    if (_forceR || _workflow.RSession.IsRemote) {
+                        var t = GetRemoteDirectoryItemsAsync(directory);
+                        entries = t.WaitTimeout(500);
+                    } else {
+                        entries = GetLocalDirectoryItems(directory);
                     }
+                    entries.ForEach(e => completions.Add(e));
                 }
             } catch (IOException) { } catch (UnauthorizedAccessException) { }
 
             return completions;
         }
         #endregion
 
+        private Task<List<RCompletion>> GetRemoteDirectoryItemsAsync(string directory) {
+            return Task.Run(async () => {
+                var session = _workflow.RSession;
+                var completions = new List<RCompletion>();
+
+                try {
+                    var rPath = directory.ToRPath();
+                    var files = await session.EvaluateAsync<JArray>(Invariant($"as.list(list.files(path = '{rPath}', include.dirs = FALSE))"), REvaluationKind.Normal);
```

Comment:
$"as.list(list.files(path = {rPath.ToRStringLiteral()}, include.dirs = FALSE))"

otherwise this won't work if path has a single quote char in it.

same below